pluginlib/pluginlib-extras.cmake

```diff
@@ -35,3 +35,7 @@ if(UNIX AND NOT APPLE)
   # ament_export_libraries() because it is not absolute and cannot be found with find_library
   list(APPEND pluginlib_LIBRARIES ${FILESYSTEM_LIB})
 endif()
+
+find_package(tinyxml2_vendor REQUIRED)
+find_package(TinyXML2 REQUIRED)
+list(APPEND pluginlib_LIBRARIES ${TinyXML2_LIBRARIES})
```

Comment:
Have we just been getting lucky that pluginlib_LIBRARIES hadn't needed the TinyXML2_LIBRARIES added before because they were being brought in from somewhere else?

Reply:
It was fine the way it was, however when crosscompiling the paths in `${TinyXML2_LIBRARIES}` where absolute in the host, so the downstream packages wouldn't find them when they were crosscompiled for the target and libraries/executables didn't have all the symbols. One way to know if crosscompiled builds will be fine is to add the following as linker flags:

```cmake
if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-undefined,error")
endif()
```

this will fail the builds even if they are not crosscompiled by making sure that there are no missing symbols.

Reply:
That's what we added to https://gitlab.com/autowarefoundation/autoware.ai/common/blob/master/autoware_build_flags/cmake/autoware_build_flags-extras.cmake#L1 This code could be added to `ament_cmake_ros`, for example.